fetchai/ledger/serialisation/shardmask.py

```diff
@@ -0,0 +1,33 @@
+import hashlib
+import struct
+
+from fetchai.ledger.bitvector import BitVector
+
+
+class ShardMask:
+    @classmethod
+    def state_to_address(cls, state, contract):
+        # TODO: note circular dependency, as this will be called by contract
+        assert contract.owner, "Contract does not have an owner"
+        return "{}.{}.state.{}".format(contract.digest.to_hex(), contract.owner, state)
+
+    @classmethod
+    def resources_to_shard_mask(cls, resource_addresses, num_lanes):
```

Comment:
It would be good to have a touch unit tests for these three methods.

Reply:
Added, though could possibly be improved - I'm simply testing that known strings generate known hashes for now.